app/services/hyrax/characterization/valkyrie_characterization_service.rb

```diff
@@ -0,0 +1,133 @@
+# frozen_string_literal: true
+require 'hydra-file_characterization'
+require 'nokogiri'
+
+class Hyrax::Characterization::ValkyrieCharacterizationService
+  ##
+  # @api public
+  # @param [Hyrax::FileMetadata] metadata  which has properties to recieve characterization values
+  # @param [Valkyrie::StorageAdapter::StreamFile] source to run characterization against
+  # @param [Hash] options the options pass to characterization
+  # @option options [Hash{Symbol => Symbol}] parser_mapping
+  # @option options [Class] parser_class
+  # @option options [Symbol] ch12n_tool
+  #
+  # @return [Hash]
+  def self.run(metadata, source = nil, **options)
+    new(metadata: metadata, source: source, **options).characterize
+    Hyrax.persister.save(resource: metadata)
+  end
+
+  ##
+  # @!attribute [rw] source
+  #   @return [Valkyrie::StorageAdapter::StreamFile]
+  # @!attribute [rw] metadata
+  #   @return [Hyrax::FileMetadata]
+  # @!attribute [rw] source
+  #   @return [Valkyrie::StorageAdapter::StreamFile]
+  # @!attribute [rw] tools
+  #   @todo clarify what this is for
```

Comment:
i'm not sure i understand `@tools` or its relationship to  `:ch12n_tool`. can you help me document it?